### PR TITLE
feat: add course summary v2

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2494](https://openedx.atlassian.net/browse/PROD-2494)
Adds a new component for course summary.

image

Screenshot 2021-09-30 at 4 55 50 PM

### CODE REVIEW

File: src/users/courseSummary/v2/CourseSummary.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  const courseSummaryRef = useRef(null);
+
+  useEffect(() => {
+    setCourseSummaryData(null);
+  }, [courseUUID]);
+
+  useEffect(() => {
+    clear('course-summary');
+    if (courseSummaryData === null) {
+      setCourseSummaryData(null);
```
Comment (DawoudSheraz) [inline src/users/courseSummary/v2/CourseSummary.jsx]: If data is already null, why set it to null again?

File: src/users/courseSummary/v2/CourseSummary.test.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  it('Default component render with Modal', () => {
+    const dataRows = wrapper.find('table.course-summary-table tbody').first().children();
+    expect(dataRows.length).toEqual(5);
+
+    let courseSummaryModal = wrapper.find('Modal#course-summary');
+    expect(courseSummaryModal.prop('open')).toEqual(true);
+
+    wrapper.find('button.btn-link').simulate('click');
+    courseSummaryModal = wrapper.find('Modal#course-summary');
+    expect(courseSummaryModal.prop('open')).toEqual(false);
```
Comment (DawoudSheraz) [inline src/users/courseSummary/v2/CourseSummary.test.jsx]: verify the existence of the sub-table with the correct data.

File: src/users/courseSummary/v2/CourseSummary.jsx (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/users/courseSummary/v2/CourseSummary.jsx]: Removed.

File: src/users/courseSummary/v2/CourseSummary.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import UserMessagesContext from '../../../userMessages/UserMessagesContext';
+import AlertList from '../../../userMessages/AlertList';
+import { getCourseData } from '../../data/api';
+import TableV2 from '../../../components/Table';
+
+export default function CourseSummary({
+  courseUUID, closeHandler,
+}) {
+  const { add, clear } = useContext(UserMessagesContext);
+  const [courseSummaryErrors, setCourseSummaryErrors] = useState(false);
```
Comment (DawoudSheraz) [inline src/users/courseSummary/v2/CourseSummary.jsx:23]: I think we can remove the show summary error state. AlertList handles this pretty well automatically if new errors related to the summary are added in the context.

**Review by DawoudSheraz — APPROVED** [review]

File: src/users/courseSummary/v2/CourseSummary.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/courseSummary/v2/CourseSummary.jsx:23]: Oh, that state is to make sure the loading icon is removed upon an error message.
Screenshot 2021-10-04 at 1 26 03 PM

**Review by azanbinzahid — APPROVED** [review]

**Pull request merged**